pkg/client/typed/discovery/discovery_client.go

```diff
@@ -123,7 +123,7 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *metav1.APIGroupList, err
 	v := &metav1.APIVersions{}
 	err = d.restClient.Get().AbsPath(d.LegacyPrefix).Do().Into(v)
 	apiGroup := metav1.APIGroup{}
-	if err == nil {
+	if err == nil && len(v.Versions) != 0 {
```

Comment:
We should check `len(v.Versions) != 0` before line 145 as well. We don't want to append an empty apiGroup to the apiGroupList.Groups.

This entire file is moved to staging/src/k8s.io/client-go/discovery/discovery_client.go